tests/client/db.rs

```diff
@@ -62,32 +67,37 @@ fn list_collections() {
 
     let results = cursor.next_n(5).unwrap();
 
-    let v3_0 = cfg!(feature = "mongodb_3_0");
-    let result_size = if v3_0 { 3 } else { 2 };
+    let db_version = db.version().unwrap();
+    let v3_1 = db_version.major <= 3 && db_version.minor <= 1;
+
+    let result_size = if v3_1 { 3 } else { 2 };
     assert_eq!(result_size, results.len());
 
-    if v3_0 {
+    if v3_1 {
         match results[0].get("name") {
             Some(&Bson::String(ref name)) => assert_eq!("system.indexes", name),
             _ => panic!("Expected BSON string!"),
         };
     }
 
+    let db_1 = if v3_1 { "test" } else { "test2" };
```

Comment:
Minor style nitpick (feel free to ignore if you disagree): I think `db1/db2` looks better than `db_1/db_2`